pep-0517.txt

```diff
@@ -166,6 +166,43 @@ Where the ``build-backend`` key exists, it takes precedence over
 Projects may still wish to include a ``setup.py`` for compatibility
 with tools that do not use this spec.
 
+This PEP also defines a ``python-paths`` key for use in ``pyproject.toml``, see
+the "In-Tree Build Backends" section below. This key would be used as follows::
+
+    [build-system]
+    # Defined by PEP 518:
+    requires = ["flit"]
+    # Defined by this PEP:
+    build-backend = "local_backend"
+    python-paths = ["backend"]
+
+
+Build Requirements
+==================
+
+This PEP places a number of additional requirements on the "build requirements"
+section of ``pyproject.toml``. These are intended to ensure that projects do
+not create impossible to satisfy conditions with their build requirements.
+
+- Project build requirements will define a directed graph of requirements
+  (project A needs B to build, B needs C and D, etc.) This graph MUST NOT
+  contain cycles.  If (due to lack of co-ordinaton between projects, for
+  example) a cycle is present, front ends MAY refuse to build the project.
+- Front ends SHOULD treat build requirements available as wheels as having
+  no build requirements - however front ends MAY have modes where they do
```

Comment:
I'm going to get nitpicky over the wording here. A requirement being available as a wheel doesn't mean it has no build requirements, it means that the frontend may choose not to build that requirement.

Suggestion:

"Where build requirements are available as wheels, front ends SHOULD use these where practical, to avoid deeply nested builds. However, front ends MAY have..."

Reply:
Thanks, that reads much better. I'll change this.